service/spi/src/test/java/org/kaazing/gateway/service/messaging/collections/MemoryTopicTest.java

```diff
@@ -16,22 +16,28 @@
 package org.kaazing.gateway.service.messaging.collections;
 
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
 
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicBoolean;
 
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.RuleChain;
 import org.kaazing.gateway.service.collections.MemoryCollectionsFactory;
+import org.kaazing.gateway.util.scheduler.SchedulerProvider;
 import org.kaazing.test.util.ITUtil;
 
 import com.hazelcast.core.ITopic;
 import com.hazelcast.core.MessageListener;
 
+
+// TODO move this test to org.kaazing.gateway.server.context.resolve (standalone/default)
+//
+// Tests not covered:
+//    - topic publish will happen-before calling add/remove inside a message listener
```

Comment:
I am not sure it's worth supporting calling publish, add and remove listener inside a message listener. We will not need this. Does Hazelcast even support it? If you want you can just throw UnsupportedOperationException and log an error.